src/client/AccountsClient.js

```diff
@@ -48,13 +53,75 @@ export class AccountsClient {
       middleware,
     });
   }
-
   getState(): Map<string, any> {
     return this.store.getState().get('accounts');
   }
   user(): UserObjectType {
     return this.getState().get('user').toJS();
   }
+  tokens(): TokensType {
+    return {
+      accessToken: localStorage.getItem(ACCESS_TOKEN),
+      refreshToken: localStorage.getItem(REFRESH_TOKEN),
+    };
+  }
+  clearTokens() {
+    localStorage.removeItem(ACCESS_TOKEN);
+    localStorage.removeItem(REFRESH_TOKEN);
+  }
+  clearUser() {
+    this.store.dispatch(clearUser());
+  }
+  async resumeSession(): Promise<void> {
+    const { accessToken } = this.tokens();
+    if (accessToken) {
+      try {
+        const decodedAccessToken = jwtDecode(accessToken);
+        const currentTime = Date.now() / 1000;
+        if (decodedAccessToken.exp < currentTime) {
+          // Access token is expired, try to request a new token pair
+          await this.refreshSession();
+        } else { // Access token is still valid, resume the session
+          this.store.dispatch(setUser(decodedAccessToken.data.user));
```

Comment:
When the access token is created the essential user data "id, username, email" are encoded within it and returned to the client. This data is also stored server side.